### PR TITLE
fix: add classes to allow V2 precision

### PR DESCRIPTION
**Pull request by SMillerDev**
Should allow this library to be used with InfluxDB v2 in v1 mode.

### CODE REVIEW

File: src/InfluxDB/Precision/InfluxDBV1Precision.php [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+namespace InfluxDB\Precision;
+
+class InfluxDBV1Precision extends InfluxDBPrecision
+{
+    /**
+     * Precision constants
+     */
+    public const PRECISION_NANOSECONDS = 'n';
+    public const PRECISION_MICROSECONDS = 'u';
+    public const PRECISION_MILLISECONDS = 'm';
```
Comment (pprkut) [inline src/InfluxDB/Precision/InfluxDBV1Precision.php]: that's the `MINUTES` value though

File: src/InfluxDB/Database.php [hunk 2]
```diff
@@ -47,14 +76,15 @@ class Database
      *
      * @throws \InvalidArgumentException
      */
-    public function __construct($name, Client $client)
+    public function __construct($name, Client $client, InfluxDBPrecision $precision = NULL)
```
Comment (pprkut) [inline src/InfluxDB/Database.php:81]: `@param` needs to be added, and the parameter type should be nullable?

File: src/InfluxDB/Database.php [hunk 3]
```diff
@@ -30,14 +33,40 @@ class Database
...
-     * Precision constants
+     * @var InfluxDBV2Precision|InfluxDBV1Precision
      */
-    const PRECISION_NANOSECONDS = 'n';
-    const PRECISION_MICROSECONDS = 'u';
-    const PRECISION_MILLISECONDS = 'ms';
-    const PRECISION_SECONDS = 's';
-    const PRECISION_MINUTES = 'm';
-    const PRECISION_HOURS = 'h';
+    protected $precision;
```
Comment (pprkut) [inline src/InfluxDB/Database.php]: This should probably match what we expect as input from the constructor.
Either `InfluxDBPrecision` or `InfluxDBV1Precision|InfluxDBV2Precision`

File: src/InfluxDB/Database.php (hunk 2 above)
Comment (SMillerDev) [reply src/InfluxDB/Database.php:81]: No, we want the default to be null. But if you pass it it should be a precision class.

Comment (SMillerDev) [reply src/InfluxDB/Database.php:81]: No, we want the default to be null. But if you pass it it should be a precision class.

**Review by pprkut — APPROVED** [review]

**Comment by SMillerDev** [comment]
Wait, should this go into master?

**Comment by pprkut** [comment]
The release branch should be fine for now. We can rebase everything onto the latest master after

**Pull request merged**